lib/domain/usecase/search/search_interactor.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:dartz/dartz.dart';
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/domain/app_state/search/search_interactor_state.dart';
+import 'package:fluffychat/domain/model/contact/contact_query.dart';
+import 'package:fluffychat/domain/model/extensions/contact/contact_extension.dart';
+import 'package:fluffychat/domain/model/extensions/contact/presentation_contact_list_extension.dart';
+import 'package:fluffychat/domain/repository/contact_repository.dart';
+import 'package:fluffychat/presentation/extensions/room_extension.dart';
+import 'package:fluffychat/presentation/extensions/room_list_extension.dart';
+import 'package:fluffychat/presentation/model/presentation_search.dart';
+import 'package:fluffychat/utils/matrix_sdk_extensions/client_stories_extension.dart';
+import 'package:fluffychat/utils/matrix_sdk_extensions/matrix_locals.dart';
+import 'package:fluffychat/widgets/matrix.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:matrix/matrix.dart';
+
+class SearchContactsAndRecentChatInteractor {
+  final ContactRepository contactRepository = getIt.get<ContactRepository>();
+
+  SearchContactsAndRecentChatInteractor();
+
+  Stream<Either<Failure, GetContactAndRecentChatSuccess>> execute({
+    required BuildContext context,
+    required String keyword,
+    int? limitContacts,
+    int? limitRecentChats,
+    bool enableSearch = false
+  }) async* {
+    try {
+      if (enableSearch) {
+        final recentChat = await _searchRecentChat(context, keyword);
+        final contacts = await contactRepository.searchContact(query: ContactQuery(keyword: keyword), limit: limitContacts);
```

Comment:
did you support `load more`?

Reply:
We will update `load more` on this ticket #317